backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/service/EdcAdapterService.java

```diff
@@ -481,148 +488,140 @@ public Response getProxyPullRequest(String url, String authKey, String authCode,
-                        log.error("Failed to obtain EDR data for " + assetId + " with " + partner.getEdcUrl());
-                        return null;
+            }
+            try {
+                if (!partnerItemStockEdcUrl.equals(partner.getEdcUrl())) {
+                    log.warn("Diverging EDC URL's for partner " + partner.getBpnl());
+                    log.warn("Edc URL from AAS " + partnerItemStockEdcUrl);
+                    log.warn("General Edc Url of partner " + partner.getEdcUrl());
+                }
+                // Request EdrToken
+                var transferResp = initiateProxyPullTransfer(partner, itemStockContractId, assetId, partnerItemStockEdcUrl);
+                String transferId = transferResp.get("@id").asText();
+                for (int i = 0; i < 100; i++) {
+                    Thread.sleep(100);
+                    transferResp = getTransferState(transferId);
+                    if ("STARTED".equals(transferResp.get("edc:state").asText())) {
+                        break;
                     }
-                    var edrURL = edrDto.endpoint().endsWith("/") ? edrDto.endpoint() : edrDto.endpoint() + "/";
-                    edrURL += materialNumber + "/" + direction;
-                    if (!href.startsWith(edrURL)) {
-                        log.warn("Diverging URLs in ItemStock Submodel request");
-                        log.warn("href: " + href);
-                        log.warn("URL from EDR: " + edrURL);
+                }
+                EdrDto edrDto = null;
+                // Await arrival of edr
+                for (int i = 0; i < 100; i++) {
+                    Thread.sleep(100);
+                    edrDto = edrService.findByTransferId(transferId);
+                    if (edrDto != null) {
+                        log.info("Received EDR data for " + assetId + " with " + partner.getEdcUrl());
+                        break;
                     }
-                    try (var response = getProxyPullRequest(href, edrDto.authKey(), edrDto.authCode(), new String[]{"$value"})) {
-                        if (response.isSuccessful()) {
-                            String responseString = response.body().string();
-                            failed = false;
-                            return objectMapper.readTree(responseString);
-                        }
-                    } catch (Exception e) {
-                        log.error("Error in ItemStock Submodel 2.0.0 transfer request for partner " + partner.getBpnl(), e);
+                }
+                if (edrDto == null) {
+                    log.error("Failed to obtain EDR data for " + assetId + " with " + partner.getEdcUrl());
+                    return doItemStockSubmodelRequest(mpr, direction, --retries);
+                }
+                var edrURL = edrDto.endpoint().endsWith("/") ? edrDto.endpoint() : edrDto.endpoint() + "/";
+                edrURL += materialNumber + "/" + direction;
```

Comment:
This line was never "needed" in a strict sense. The whole purpose of that line is to check if the path segments stored in the AAS are plausible in their semantics. 

And the intention was never, to override the href data in any way here.. The call to the href address from the href should be executed regardless of the outcome of that check. All it does, is leave a hint in the logs, if an unexpected divergence happens here, because then very probably something else will be going wrong just a short time thereafter ...

Fortunately, so far in all the tests I have done, this test has never actually failed. I.e. the content of the href is fully in line with the expectations :)